### PR TITLE
Feature: Added support for Lucid Link v3

### LINKED ISSUE #16586: Feature: Add support for Lucid Link v3+
### What feature or improvement do you think would benefit Files?

They have updated the file structure and no longer give a path for the cloud spaces in v3.

### Requirements

- Search `C:/Volumes` and find all system link paths to display as Lucid link drives this would work for v2 and v3 clients

### Files Version

3.8.1.0

### Windows Version

10.0.26100.2314

### Comments

_No response_

### PR DESCRIPTION
**Pull request by Lamparter**
**Resolved / Related Issues**

- Closes #16586

### CODE REVIEW

File: src/Files.App/Utils/Cloud/Detector/LucidLinkCloudDetector.cs [hunk 1]
```diff
@@ -43,5 +52,36 @@ protected override async IAsyncEnumerable<ICloudProvider> GetProviders()
...
 			}
 		}
+
+		private async IAsyncEnumerable<ICloudProvider> GetLucidLinkV3Providers(string volumePath)
+		{
+			if (Directory.Exists(volumePath))
+			{
+				foreach (string directory in Directory.GetDirectories(volumePath))
+				{
+					if (IsSymlink(directory))
```
Comment (yaira2) [inline src/Files.App/Utils/Cloud/Detector/LucidLinkCloudDetector.cs:62]: It seems like the symlink is one directory deeper, so this will return false.

**Review by Josh65-2201 — CHANGES_REQUESTED** [review]
The old method can be removed as both v2 and v3 will access it from the volumes folder. You would also need to search 1 layer sub folders for the system link. Volumes/Workspace/filespace (the system link) is how it's formatted.

**Comment by Josh65-2201** [comment]
The workspace and filespace path are backwards so it always returns not a folder

**Review by Josh65-2201 — APPROVED** [review]

File: src/Files.App/Utils/Cloud/Detector/LucidLinkCloudDetector.cs [hunk 2]
```diff
@@ -13,25 +11,34 @@ namespace Files.App.Utils.Cloud
...
+		}
 
-					string[] orgNameFilespaceName = syncFolder.Split(".");
-					string path = Path.Combine($@"{Constants.UserEnvironmentPaths.SystemDrivePath}\Volumes", orgNameFilespaceName[1], orgNameFilespaceName[0]);
-					string filespaceName = orgNameFilespaceName[0];
+		private async IAsyncEnumerable<ICloudProvider> GetProvidersFromDirectory(string directory)
+		{
+			foreach (string subDirectory in Directory.GetDirectories(directory))
+			{
+				if (IsSymlink(subDirectory))
```
Comment (yaira2) [inline src/Files.App/Utils/Cloud/Detector/LucidLinkCloudDetector.cs:36]:
```suggestion
				if (Win32Helper.HasFileAttribute(subDirectory, System.IO.FileAttributes.ReparsePoint))
```

**Review by yaira2 — COMMENTED** [review]
`private static bool IsSymlink` can be removed

**Review by yaira2 — APPROVED** [review]

**Pull request merged**